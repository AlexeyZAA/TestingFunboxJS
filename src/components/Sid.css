.list6b {
  padding:0;
  list-style: none;
}
.list6b li{
  padding:6px;
}
.list6b li:before {
  padding-right:10px;
  font-weight: bold;
  color: #C0C0C0;
  content: "\2714";
}


.lival {
  float: left;
  width: 140px;
  border-bottom: solid 1px antiquewhite;
}

.libtn {
  float: right;
}

