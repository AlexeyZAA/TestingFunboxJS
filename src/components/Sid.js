import React, { Component } from "react";
import { sortableContainer, sortableElement } from "react-sortable-hoc";
import arrayMove from "array-move";
import { Layout, Input, List, Button } from "antd";

const { Sider } = Layout;

const SortableItem = sortableElement(({ value }) => <li>{value}</li>);

const SortableContainer = sortableContainer(({ children }) => {
  return <ul>{children}</ul>;
});

class Sid extends React.Component {
  state = {
    collapsed: false,
    inputpoint: "",
    datapointlist: [],
    items: ["Item 1", "Item 2", "Item 3", "Item 4", "Item 5", "Item 6"],
  };

  onSortEnd = ({ oldIndex, newIndex }) => {
    this.setState(({ items }) => ({
      items: arrayMove(items, oldIndex, newIndex),
    }));
  };

  toggle = () => {
    this.setState({
      collapsed: !this.state.collapsed,
    });
  };

  handleEnter = event => {
    if (event.keyCode === 13) {
      this.setState({
        inputpoint: event.target.value,
      });
      this.state.datapointlist.push({
        title: event.target.value,
        key: this.state.datapointlist.length,
      });
    }
  };
  inputPointClean = event => {
    if (event.keyCode === 13) event.target.value = "";
  };

  itempointDel = param => {
    let newitemarr = [];
    newitemarr = this.state.datapointlist.filter(item => {
      return item.key !== param.key;
    });

    for (let i = 0; i < newitemarr.length; i++) {
      newitemarr[i]["key"] = i;
    }
    this.setState({
      datapointlist: newitemarr,
    });

    this.setState({
      inputpoint: "",
    });
  };

  render() {
    const { items } = this.state;

    return (
      <Sider
        breakpoint="lg"
        collapsedWidth="0"
        theme="light"
        onBreakpoint={broken => {
          console.log(broken);
        }}
        onCollapse={(collapsed, type) => {
          console.log(collapsed, type);
        }}
      >
        <Input
          placeholder="Новая точка маршрута"
          onKeyUp={this.inputPointClean}
          onKeyDown={this.handleEnter}
        />

        <SortableContainer onSortEnd={this.onSortEnd}>
          {items.map((value, index) => (
            <SortableItem key={`item-${value}`} index={index} value={value} />
          ))}
        </SortableContainer>
      </Sider>
    );
  }
}

export default Sid;
