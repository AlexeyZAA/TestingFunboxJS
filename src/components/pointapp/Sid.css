.dul {
  position: relative;
  z-index: 0;
  background-color: #f3f3f3;
}

.dli {
  float: left;
  height: 30px;
  display: flex;
  align-items: center;
  width: 150px;
  padding: 0 5px;
  background-color: #fff;
  box-sizing: border-box;
  user-select: none;
  border-bottom: 1px solid #999;
  margin-left: -30px;
}

.btnD {
  float: right;
  margin-right: 3px;
}