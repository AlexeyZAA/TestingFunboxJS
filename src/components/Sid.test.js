import React from 'react'
import { render } from '@testing-library/react'
import Sid from './Sid'

/* eslint-env jest */

/** Протестируем ввывод списка точек в Сидебаре, дав ему подготовленные параметры. Формирование параметров в другом тесте */
const items = [
  {
    title: 'Точка 1',
    key: 0,
    par: {
      geometry: {
        type: 'Point',
        coordinates: [53.35475063654921, 83.77012869842511]
      },
      properties: {
        iconContent: '',
        hintContent: 'Узел маршрута',
        balloonContentHeader: 'Название узла',
        balloonContentBody: 'Точка 1',
        balloonContentFooter: 'Балун всплывайка'
      },
      options: { preset: 'islands#circleIcon', draggable: true }
    }
  },
  {
    title: 'Точка 2',
    key: 1,
    par: {
      geometry: {
        type: 'Point',
        coordinates: [53.351412638902175, 83.7630047512809]
      },
      properties: {
        iconContent: '',
        hintContent: 'Узел маршрута',
        balloonContentHeader: 'Название узла',
        balloonContentBody: 'Точка 2',
        balloonContentFooter: 'Балун всплывайка'
      },
      options: { preset: 'islands#circleIcon', draggable: true }
    }
  },
  {
    title: 'Точка 3',
    key: 2,
    par: {
      geometry: {
        type: 'Point',
        coordinates: [53.35721545073843, 83.76472136505083]
      },
      properties: {
        iconContent: '',
        hintContent: 'Узел маршрута',
        balloonContentHeader: 'Название узла',
        balloonContentBody: 'Точка 3',
        balloonContentFooter: 'Балун всплывайка'
      },
      options: { preset: 'islands#circleIcon', draggable: true }
    }
  }
]

test('тест снимок списка контрольных узлов маршута ', () => {
  const { asFragment } = render(<Sid items={items} />)
  expect(asFragment()).toMatchSnapshot()
})
