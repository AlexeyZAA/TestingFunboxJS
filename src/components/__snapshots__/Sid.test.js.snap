// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тест снимок списка контрольных узлов маршута  1`] = `
<DocumentFragment>
  <aside
    class="ant-layout-sider ant-layout-sider-light"
    style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
  >
    <div
      class="ant-layout-sider-children"
    >
      <input
        class="ant-input"
        placeholder="Новая точка маршрута"
        type="text"
        value=""
      />
      <ul
        class="dul"
      >
        <div>
          <li
            class="dli"
          >
            Точка 1
          </li>
          <button
            class="ant-btn ant-btn-primary ant-btn-circle"
            type="button"
          >
            <span>
              del
            </span>
          </button>
        </div>
        <div>
          <li
            class="dli"
          >
            Точка 2
          </li>
          <button
            class="ant-btn ant-btn-primary ant-btn-circle"
            type="button"
          >
            <span>
              del
            </span>
          </button>
        </div>
        <div>
          <li
            class="dli"
          >
            Точка 3
          </li>
          <button
            class="ant-btn ant-btn-primary ant-btn-circle"
            type="button"
          >
            <span>
              del
            </span>
          </button>
        </div>
      </ul>
    </div>
  </aside>
</DocumentFragment>
`;
